File: java/Montecarlo_s.java
import java.util.*;

public class Montecarlo_s{


    public static void main(String[] args){
	Random rand = new Random();
	double pih = 0;
	int count = 0;
	int lim = Integer.MAX_VALUE;
	double x;
	double y;
	for(int i = 0 ; i < lim; i++){
		x = rand.nextDouble();
		y = rand.nextDouble();
		if(x*x + y*y < 1){
			count++;
	    	}
	}
	pih = 4.0 * (count + 0.0)/lim;
	System.out.println(pih);

    }

}

File: c/Montecarlo_s.c

#include<stdio.h>
#include<stdlib.h>

double random_f(void){
      double r = (double)rand()/(double)RAND_MAX;
      return r;
}

int main(void){
	double count = 0.0;
	double pih = 0;
	int lim = 2147483647;
	double dlim = 2147483647.0;
	double x;
	double y;
	int i;
	for(i = 0; i < lim; i++){
		x = random_f();
		y = random_f();
		if(x*x + y*y < 1){
			count++;
		}
	}
	pih = 4.0 * count/dlim;
	printf("%lf\n", pih);
}


